src/geometry/WorkingGeometry.ts

```diff
@@ -99,28 +109,26 @@ export class WorkingGeometry {
     public bake(): BakedGeometry {
         this.combine();
 
-        const bakedVertices = this.vertices.map((workingVec: vec4) => [
+        const bakedVertices = flatMap(this.vertices, (workingVec: vec4) => [
             workingVec[0],
             workingVec[1],
             workingVec[2]
         ]);
         const bakedIndices = this.faces.reduce((accum: number[], face: Face) => {
             return accum.concat(face.indices);
         }, []);
-        const bakedNormals = this.normals.map((workingVec: vec4) => [
+        const bakedNormals = flatMap(this.normals, (workingVec: vec4) => [
             workingVec[0],
             workingVec[1],
             workingVec[2]
         ]);
-
-        // Make all of the baked shapes red for now.
-        const bakedColors = bakedVertices.map(() => [1, 0, 0]);
+        const bakedColors = flatMap(range(bakedVertices.length), () => this.fillColor.asArray());
```

Comment:
This will generate too many colours, actually. Before, bakedVertices just used map. Now, it used flatMap, so `range(bakedVertices.length)` is three times longer than we need it to be

Reply:
We used to `flatten` `bakedColors` [here](https://github.com/calder-gl/calder/pull/86/files/d5d62f0350619b9800a3b5e8605c19dc4a2d51e0#diff-91134b308a945178c7edfef38170b1dcL123), is this not equivalent?

Reply:
The flatmap I'm talking about is the one in `bakedVertices`, not `bakedColors`.

Before, vertices could have been `[[1, 0, 0], [0, 1, 0]]`, which maybe gets mapped to `[[0, 0, 1], [0, 0, 1]]` if we make it blue. Then, that gets flat mapped to `[0, 0, 1, 0, 0, 1]`.

Now, since vertices are flatmapped to begin with, vertices would look like `[1, 0, 0, 0, 1, 0]`, which means when we flatmap to generate colours, we get `[0, 0, 1, 0, 0, 1, 0, 0, 1, 0, 0, 1, 0, 0, 1, 0, 0, 1,]` because each entry in vertices generates three entries in colours.

Reply:
If we do `range(this.vertices.length)` instead of `bakedVertices.length` it should have the right number I think

Reply:
Ahh, gotcha! Good call! I see now. 👍